### DISCUSSION TITLE
chrome.exe "zombie process" not terminated (Win 11, GoogleChromePortable)

### DISCUSSION THREAD

**Discussion by emp-00** [gildas-lormeau/single-file-cli#97] [Q&A] [answered]
Boundary conditions:
- Running single-file-cli v2.0.43 on Win11 in combination with [Google Chrome Portable](https://portableapps.com/apps/internet/google_chrome_portable)
- Saved single-file.exe in this directory: **E:\Single-File-cli_ChromePortable**
- Google Chrome Portable installed in E**:\Single-File-cli_ChromePortable\GoogleChromePortable**
- No other chrome.exe processes running, I'm using Vivaldi, Brave and FIrefox (running) but no other "real" Chrome browsers on my system, I only used the Portable chrome for my command line scripting project

Command line (reproducible example, happens with any website): 
`single-file --browser-executable-path="D:\Single-File-cli_ChromePortable\GoogleChromePortable\GoogleChromePortable.exe" --browser-arg="--user-data-dir=D:\Single-File-cli_ChromePortable\GoogleChromePortable\Data\profile" "https://www.wikipedia.com" wikipedia.html`

Symptoms:
- In principle. the script runs fine, wikipedia.html is correctly downloaded and saved, no error message displayed
- However, two **chrome.exe** "zombie processes" are remaining in running state, they are NOT terminated
- Looking at C:\temp\UserTemp directories such as **"nsr3E07.tmp"** containing two files **"KillProc.dll"** and **"registry.dll"** are also remaining: the temporary directory and the two files are NOT deleted
- When calling single-file-cli multiple times, each two more zombie processes and temporary directories/files are remaining
- Additionally, with the 2nd call chrome is not using the desired profile any more (Default under Data\profile) due to the already running and conflicting zombie processes

Please try to reproduce, here it's 100% reproducible - also after system restart. Maybe it's caused by using Google Chrome Portable? Is the portable version not completely compatible with single-file-cli? I need to use a separate chrome(ium) browser with a separate profile for my project, that's why I chose Chrome Portable. 

Currently I need to manually terminate the chrome.exe zombie processes (and the temporary files piling up) - would be great if this can be fixed. In case my command line is causing the issue, any help is appreciated! Thanks very much for the great tool, other than this, it's working perfectly!

**Comment by emp-00** [comment, marked as answer]
I found the solution! It's required to use the downstream path to the actual **"chrome.exe"** of the Portable Chrome and NOT **"GoogleChromePortable.exe"** ! Maybe it's worthwhile adding this to the FAQ?

This works perfectly, it uses my profile as desired and no zombie processes and no temporary files remain - THANKS for the great tool, now my project can fly :-)

`single-file --browser-executable-path="D:\Single-File-cli_ChromePortable\GoogleChromePortable\App\Chrome-bin\chrome.exe" --browser-arg="--user-data-dir=D:\Single-File-cli_ChromePortable\GoogleChromePortable\Data\profile" "https://www.wikipedia.com" wikipedia.html`

**Comment by gildas-lormeau** [comment]
Thanks for the additional information. I'm glad to hear you found the cause of the problem.

In fact, there is no FAQ specific to SingleFile CLI yet. I'll move your bug report to the "Discussions" tab to make it visible to people.
